antora/docs/modules/ROOT/pages/trusting_bundles.adoc

```diff
@@ -109,3 +109,58 @@ In order to add a new task to the existing set of Red Hat acceptable tasks, the
 ** You may view the acceptable bundles xref:{acceptable-bundles}[here].
 ** Additionally, you can view the task bundle in the https://quay.io/redhat-appstudio-tekton-catalog[Red Hat AppStudio Tekton Catalog]
+
+== Adding a custom task to the trusted task list
+
+Some users may want to add a custom build task to the default acceptable bundles list without pushing that task to the upstream acceptable task list. Below is a possible approach to make that happen.
+
+EC requires a customized policy in order to accept a custom task. Such a policy must include an additional data source with task bundle refs and digests relevant to the new custom task.
+
+For example, starting with https://github.com/enterprise-contract/config/blob/main/redhat-no-hermetic/policy.yaml[Red Hat's non-hermetic build policy], a new data source can be added under the `sources.data` key. See https://github.com/simonbaird/securesign-ec-config/blob/main/policy.yaml#L24[this policy] for a complete example.
+
+[,bash]
+---
+name: Secure Sign Custom Enterprise Contract Policy
+
+description: >-
+  Based on the standard Red Hat (non hermetic) policy
+  (see https://github.com/enterprise-contract/config/blob/main/redhat-no-hermetic/policy.yaml)
+  but with one additional data source. Currently this is an experimental POC,
+  see https://issues.redhat.com/browse/EC-147 .
+
+publicKey: "k8s://openshift-pipelines/public-key"
+
+sources:
+  - name: Default with extra source
+    policy:
+      - github.com/enterprise-contract/ec-policies//policy/lib
+      - github.com/enterprise-contract/ec-policies//policy/release
+
+    data:
+      - oci::quay.io/redhat-appstudio-tekton-catalog/data-acceptable-bundles:latest
+      - github.com/release-engineering/rhtap-ec-policy//data
+
+      # This is the one additional data source
+      # See https://github.com/simonbaird/securesign-ec-config/tree/main/policy-data
+      - github.com/simonbaird/securesign-ec-config//policy-data?ref=main
+
+    config:
+      include:
+        - '@redhat'
+      exclude:
+        - hermetic_build_task
+        - tasks.required_tasks_found:prefetch-dependencies
+----
+
+That custom data source should be kept updated as new Tekton task bundles are pushed, perhaps via some automation or using https://enterprisecontract.dev/docs/ec-cli/main/ec_track_bundle.html[`ec track bundle`].
+
+The customized policy can be provided to EC with the `--policy` parameter, and should allow custom tasks to validate using the new data source.
+
+[,bash]
+----
+$ export IMAGE=quay.io/redhat-user-workloads/rhtas-tenant/rhtas-stack-1-0-beta/rekor-server@sha256:7fb224c62e5ea5b43ad123529487f07553626e9cd1f908d515114ee414679d6b
+
+$ export POLICY=github.com/simonbaird/securesign-ec-config
+
+$ ec validate image --image $IMAGE --policy $POLICY --public-key <(cat key.pub) --ignore-rekor --output data=data.yaml --output policy-input=policy-input.json --output yaml | yq .components[].violations
```

Comment:
I was wondering about the extra data and policy-input flags here - technically they're not needed and not 100% related to this process, but perhaps there's no harm to leave them in there. Maybe you could mention them and explain they could be used for further troubleshooting if required.

Reply:
I added a quick note about the flags at the end.